src/partner-access/partner-access.service.ts

```diff
@@ -80,11 +87,18 @@ export class PartnerAccessService {
       .getMany();
   }
 
-  async assignPartnerAccessOnSignup(
-    partnerAccessCode: string,
-    userId: string,
-  ): Promise<PartnerAccessEntity> {
-    const partnerAccess = await this.getValidPartnerAccessCode(partnerAccessCode);
+  async assignPartnerAccessOnSignup({
+    partnerAccessCode,
+    userId,
```

Comment:
The method params are confusing - it's not clear what combination is valid. What do you think about having two methods - the original one and then one that's say "assignAutomaticPartnerAccessOnSignup" which handles assignment with partner id?